src/freenet/client/filter/WebPFilter.java

```diff
@@ -0,0 +1,137 @@
+package freenet.client.filter;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Map;
+
+import freenet.support.Logger;
+import freenet.support.Logger.LogLevel;
+
+public class WebPFilter extends RIFFFilter {
+
+	protected final int AnimationFlag = (1 << 6);
+	@Override
+	public byte[] getChunkMagicNumber() {
+		return new byte[] {'W', 'E', 'B', 'P'};
+	}
+	
+	class WebPFilterContext {
+		public int VP8XFlags = 0;
+		public boolean hasVP8 = false;
+		public boolean hasVP8L = false;
+		public boolean hasANIM = false;
+	}
+	@Override
+	public Object createContext() {
+		return new WebPFilterContext();
+	}
+
+	@Override
+	public boolean readFilterChunk(byte[] ID, int size, Object context, DataInputStream input, DataOutputStream output, String charset, Map<String, String> otherParams,
+			String schemeHostAndPort, FilterCallback cb) throws DataFilterException, IOException {
+		boolean logMINOR = Logger.shouldLog(LogLevel.MINOR, this.getClass());
+		WebPFilterContext ctx = (WebPFilterContext)context;
+		if(ID[0] == 'V' && ID[1] == 'P' && ID[2] == '8' && ID[3] == ' ') {
+			// VP8 Lossy format: RFC 6386
+			// Most WebP files just contain a single chunk of this kind
+			if(ctx.hasVP8 || ctx.hasVP8L || ctx.hasANIM) {
+				throw new DataFilterException("WebP error", "WebP error", "Unexpected VP8 chunk was encountered");
+			}
+			output.write(ID);
+			if(logMINOR) Logger.minor(this, "Passing through WebP VP8 block with " + size + " bytes.");
+			VP8PacketFilter VP8filter = new VP8PacketFilter(true);
+			CodecPacket filteredPacket = VP8filter.parse(readAsCodecPacket(input, size));
+			size = filteredPacket.payload.length;
+			output.writeInt(((size & 0xff000000) >> 24) | ((size & 0x00ff0000) >> 8) | ((size & 0x0000ff00) << 8) | ((size & 0x000000ff) << 24));

```

Comment:
We cannot all of a sudden emit less bytes than were specified in the top-level RIFF header. Any missing bytes must be accounted for, for instance by appending a JUNK chunk to the end of the file (if the number of skipped bytes allows for that). Chunk padding should also be taken into account when the chunk size ends up being an odd number.

In short, as long as it is not truly needed on packet level, having this flexibility is not yet worth the effort.